lib/shopify_api/plugs/auth_shop_session_token.ex

```diff
@@ -18,22 +30,47 @@ defmodule ShopifyAPI.Plugs.AuthShopSessionToken do
 
   require Logger
 
-  def init(opts), do: opts
+  alias ShopifyAPI.App
+  alias ShopifyAPI.AppServer
+  alias ShopifyAPI.AuthTokenServer
+  alias ShopifyAPI.ShopServer
+  alias ShopifyAPI.UserTokenServer
+
+  @defaults [use_online_tokens: false]
 
-  def call(conn, _options) do
+  def init(opts), do: Keyword.merge(@defaults, opts)
+
+  def call(conn, options) do
     with ["Bearer " <> token] <- get_req_header(conn, "authorization"),
          {:ok, app} <- get_app_from_token(token),
          {true, jwt, _jws} <- verify_token(token, app.client_secret),
-         {:ok, my_shopify_domain} <- get_my_shopify_domain_from_jwt(jwt),
-         {:ok, shop} <- ShopifyAPI.ShopServer.get(my_shopify_domain),
-         {:ok, auth_token} <- ShopifyAPI.AuthTokenServer.get(my_shopify_domain, app.name) do
+         {:ok, myshopify_domain} <- get_my_shopify_domain_from_jwt(jwt),
+         {:ok, user_id} <- get_user_from_jwt(jwt),
+         {:ok, shop} <- ShopServer.get(myshopify_domain),
+         {:ok, auth_token} <- AuthTokenServer.get(myshopify_domain, app.name),
+         {:ok, user_token} <-
+           fetch_user_token(myshopify_domain, app.name, user_id, options[:use_online_tokens]) do
       conn
       |> assign(:app, app)
       |> assign(:shop, shop)
       |> assign(:auth_token, auth_token)
+      |> assign(:user_id, user_id)
+      |> assign(:user_token, user_token)
     else
+      {:error, {:user_token_not_found, %{message: message, app: app_name, shop: shop}}} ->
+        Logger.debug("Could not authenticate user #{inspect(message)}")
+        {:ok, app} = AppServer.get(app_name)
+
+        install_url = App.install_url(app, shop, options[:use_online_tokens])
+        Logger.debug("install_url: #{inspect(install_url)}")
+
+        conn
+        |> resp(403, "Not Authorized.")
+        |> put_resp_header("X-Shopify-API-Request-Failure-Reauthorize-URL", install_url)
```

Comment:
And then this header has to be set up in the `authenticatedFetch` options 🤯 .

Is this even documented ANYWHERE in Shopify's docs? I couldn't find anything. I had to do digging in the source code to even see what the options can be.

How do we document that this is the necessary setup?